examples/gp.py

```diff
@@ -1,23 +1,20 @@
-import matplotlib
-matplotlib.use('Agg')  # noqa: E402
-import matplotlib.pyplot as plt
-import jax
-
 import argparse
 
+import matplotlib
+import matplotlib.pyplot as plt
 import numpy as onp
-from jax import vmap
+
+import jax
 import jax.numpy as np
 import jax.random as random
+from jax import vmap
 
 import numpyro.distributions as dist
 from numpyro.handlers import sample
 from numpyro.hmc_util import initialize_model
 from numpyro.mcmc import mcmc
 
-from jax.config import config
-# we use double precision to minimize any possible numerical instabilities in jax linear algebra
-config.update('jax_enable_x64', True)
```

Comment:
On my system too, while the posterior estimates look similar, the predictions are very off and are likely affected by lack of precision (attached image). I haven't really investigated further, but we should revert this until we understand what the issue is.

[gp_plot (2).pdf](https://github.com/pyro-ppl/numpyro/files/3254975/gp_plot.2.pdf)

Reply:
I just discover that I use an old version of `jaxlib` (0.1.14), and get a nice plot. Then when I update it to `0.1.17`, I get a plot similar to you. This might be a regression of `jaxlib`.

Reply:
And when I update `jax` to `0.1.36`, I get a nice plot again. >"<

Reply:
Maybe this [PR](https://github.com/google/jax/pull/785) is helpful for MVN because during warmup phase, potential_energy will be `nan` when something is wrong (instead of some random stuffs). It is just my assumption though. Will test on GPU too.

Reply:
no idea what's going on but i think it may actually be a problem in prediction not in inference

Reply:
Yup, you are right. It seems that the issue is from getting some `nan` over some of the posterior samples. I just replace `percentage` with `nanpercentage` and get good plot. :D

Reply:
another "solution" for single precision seems to be to increase the jitter a lot (e.g to 0.01)

Reply:
hah, so we have 3 solutions so far:
+ update jax
+ increase jitter
+ use `nanpercentage`

I would go for `nanpercentage` to be safe for the case posterior contains some unstable samples.

Reply:
actually i think the best solution is the make the following change:

``` sigma_noise = np.sqrt(np.clip(np.diag(K), 0.0, None)) * jax.random.normal(rng, (X_test.shape[0],))```

then you can use a small jitter no problem

Reply:
(the nan's are from taking the sqrt of negative reals)

Reply:
Yup, you are right. I can recall that phenomenon in Pyro. Despite that theoretically, `k_pp - np.matmul(k_pX, np.matmul(K_xx_inv, np.transpose(k_pX)))` will have positive diagonal part, there is no such guarantee for numerical computers. I will use your solution instead.